### PR TITLE
feat(errors): Beautify stack traces, handle in app

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Make stack traces (somewhat) nicer, with support for in app frames

https://github.com/user-attachments/assets/aeb91191-a9cd-4a3a-8244-af1b3236222a

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: frontend/src/lib/components/Errors/ErrorDisplay.tsx [hunk 1]
```diff
@@ -63,15 +66,32 @@ function StackTrace({ rawTrace }: { rawTrace: string }): JSX.Element | null {
 }
 
 function ChainedStackTraces({ exceptionList }: { exceptionList: ExceptionTrace[] }): JSX.Element {
+    const [showAllFrames, setShowAllFrames] = useState(false)
```
Comment (daibhin) [inline frontend/src/lib/components/Errors/ErrorDisplay.tsx:69]: Should this be hoisted up to the logic or somewhere? I could imagine it's pretty annoying to have to toggle it on every exception you view if that's your intention

Comment (neilkakkar) [reply frontend/src/lib/components/Errors/ErrorDisplay.tsx:69]: I wasn't sure, but in general I wouldn't _expect_ so, the full stack trace is very rarely useful, hence all our effort to figure out what's in app and what's not. I think the only time I toggled this in sentry was when trying to see what all they offer here 😅  (so never for debugging in 3yrs). But no strong feelings, happy to persist in a logic as well

**Review by daibhin — APPROVED** [review]
🚢

File: frontend/src/lib/components/Errors/ErrorDisplay.tsx (hunk 1 above)
Comment (daibhin) [reply frontend/src/lib/components/Errors/ErrorDisplay.tsx:69]: Let's go with it as is and let customers tell us if they want something else

Comment (daibhin) [reply frontend/src/lib/components/Errors/ErrorDisplay.tsx:69]: Another perfectly reasonable thing here is to just exclude it 🤷 If neither of us have ever needed it begs the question whether we want it as a feature

Comment (neilkakkar) [reply frontend/src/lib/components/Errors/ErrorDisplay.tsx:69]: reasonable, I'll ship it and add an action to track how often its used :P

**Pull request merged**